2024/planning-process.md

```diff
@@ -0,0 +1,187 @@
+4. Focus Area proposals must:
+    1. Name a specific web technology.
+        1. This can be a web technology that has not yet been implemented in any or all browser engines.
+        2. Or this can be a web technology that has already been implemented in all three engines, but those implementations have multiple test failures that reveal a lack of compliance to the web standard, causing a lack of interoperability.
+    2. The feature must be defined in a sufficiently mature standards-track web specification from one the following organizations. Please provide link to the web standard.
+        1. [IETF](https://www.ietf.org/) (Standard Track)
+        2. [Khronos Group](https://www.khronos.org/)
+        3. [TC39](https://tc39.es/) (Stage 2+)
+        4. [W3C](https://www.w3.org/) (Recommendation Track)
+        5. [WHATWG](https://whatwg.org/)
+    3. Describe the benefit to web developers and the web itself
+    4. Link to existing tests (if any) and/or describe plan to create needed tests
+5. Investigation Project proposals must:
+    1. Outline their scope and their goals,
+    2. Not substantially be work better suited to a group within a standards development organization,
+    3. Clearly provide present or future impact on cross-browser interoperability.
+
+As soon as each proposal is submitted as a GitHub issue, discussion of that proposal can begin. Questions can be asked. The proposal can be edited and refined.
+
+At the end of the three weeks, the deadline will end the period when brand new proposals can be submitted.
+
+A fourth week will allow for continued discussion and refinement. The deadline at the end of the fourth week ends the opportunity to significantly change the scope of the proposal.  The proposal will be evaluated on its state at that time.
+
+
+## Focus Area Selection
+
+### Elimination Round 1 : Disqualification
+
+After the Call of Proposals Period has ended, the organizing committee will finalize an official list of all Focus Area proposals.
+
+Any Focus Area proposal that does not meet the minimum publicly-stated selection criteria will be considered for preliminary elimination. For example:
+
+1. The proposal is overly vague or broad.
+2. The proposed technology is already interoperable according to WPT results.
+3. The proposed technology is not on a standards track, or its standard is not sufficiently mature.
+
+Decisions will be made at a specific meeting for discussion of preliminary eliminations. Any member organization can bring up a proposal they believe should be eliminated. Brief debate can be had. If **there is quick consensus**, that proposal will be eliminated. If there is not consensus and a longer debate is needed, the proposal will be not be eliminated and will instead move on to the next round.
```

Comment:
It's unclear to me what the value of not eliminating if anyone proposes elimination is. Or at least: if someone makes it clear that they will veto a proposal in later rounds it may as well be eliminated early.

Reply:
> I think it seems pretty easy to present, have the group accept that and move on

Nevertheless this will take time, and it's hard enough to make decisions already. The purpose of rounds 1 and 2 is about efficiency of easy decisions, so we should stick to easy decisions and not ones where there is any debate at all.

Reply:
Yeah, as @chrishtr says—it's about making non-controversial decisions quickly.

It's also intended that the first round doesn't have too many subjective decisions—a spec either does or does not meet the criteria we've set (and I hope we don't get into disagreements here!), tests already show interoperability or they don't (we can of course get into a debate about whether the final 5% matters!), though "overly vague or broad" is much more so subjective.

@jgraham: was your original question intended to be "why not let people veto up front, for any reason"? if so—the current kinda short list of reasons for disqualification are there because they're largely what we practically did in prior years, and because things that are largely objective are likely easier to get consensus on.

Reply:
Is it possible that vagueness is causing us to be talking about completely different things?  I think yes let's ask people to 'veto' here, and not at the end.  That is what I am saying - I might be completely alone in this... But honestly, I also kind of dislike that we're even using words like "vote" and "veto" at all because they conjure up something that I don't honestly think is really happening here.  I understand why, it's a kind of vote - but really what we're doing is more like using some joint worksheets to do complicated sorting and planning and negotiate (with what little power any of us actually has to do so) some plausible subset of things we all agree we could work on improving interoperability of.

Let's say two concrete examples:

* Somebody submits that we should work on interoperability of the `is=""` attribute.  It's in the WHATWG HTML spec, which I think has always had the same definition "two in favor, none opposed".  Except Apple _is_ opposed, and has a public position many times stating as much, including a recent WebKit standards position.  So... Is it disqualified in that case? Or does this require a "veto"?  If the latter, we can say it now, why drag it out.  It's just a particularly easy example, but it's not unique: People are going to submit things that have negative standards positions - is this the venue to debate it? I kind of think not.

* Somebody submits we should work on 'all the color stuff' a few years ago and it's pointed out that this requires much low level rework in order to actually bring the value, so it's not likely something that would actually improve this year and there are lots of _other_ things where we think we could all agree and get stuff measurably done and shipped --- but hey we are actually working on all that low level stuff and totally support it!  That isn't currently disqualified, and it's certainly not a "veto" - but it's a known known and it seems to me the group could choose to say "that seems totally reasonable, let's just exclude it" and doing that as early as possible just seems good... To me.

Reply:
Let's say someone proposes "Geolocation Sensor" for Interop 2024. It has a [spec](https://w3c.github.io/geolocation-sensor/) on a standards track at W3C, and [tests](https://github.com/web-platform-tests/wpt/tree/master/geolocation-sensor). So per the criteria given we can't eliminate the proposal at this stage. But it also has [multiple](https://github.com/WebKit/standards-positions/issues/32) [negative](https://github.com/mozilla/standards-positions/issues/36) standards positions, and no evidence of implementations going ahead despite those standards positions.

It seems like a pretty safe bet to me that proposal is not going to reach consensus.

But per the process here, we have to take it forward in case we want to have a longer debate at some later stage. But it's not clear to me what form that debate could possibly take, and indeed debating proposals as a group isn't part of any of the later stages of the proposed process. So assuming at least one org is in favour in round 2, the proposal has to go all the way through to round 3, even though we had the information in round 1 to know it wasn't going to reach consensus (because a participant could have said so).

Obviously this isn't the most serious concern; eventually we'll incorporate the information that the proposal isn't going to reach consensus, and it won't get selected. But it is an inefficiency in a process that claims to be optimising for making the best use of everyone's time.